### ISSUE TITLE
Wrong detection in javascript:S3854 (invoke super before this)

### ISSUE THREAD

**Issue by AMorgaut** [SonarSource/SonarJS#1362]
**I want to report a bug.**

**SonarJS version:** 5.2.1<br>
**SonarQube version:** Community Edition Version 7.8 (build 26217)<br>
**SonarLint version:** 4.1.1.3345<br>

**Rule key:** javascript:S3854

**Reproducer** 
```
class Foo extends Bar {
    constructor(handler) {
        function handlerDecorator(...args) {
            this.baz = new Bar();
            handler(...args);
        }
        super(handlerDecorator); // False-positive here
    }
}
```
**Expected behavior**

No raised error. `this` is not modified before `super()` is called, it will eventually be modified when the param will be invoked.

**Comment by yassin-kammoun-sonarsource** [comment]
Fixed with #2279.
